docs: merge Alibaba install guide into quick install guide (#21581)
cilium-team/cilium: Documentation/gettingstarted/k8s-install-default.rst

       .. include:: ../installation/rancher-desktop-configure.rst

    .. group-tab:: Alibaba ACK

        .. include:: ../beta.rst

        .. note::

>             The AlibabaCloud ENI integration is still subject to below limitations:
> 
>             - The Alibaba ENI integration of Cilium is currently only enabled for IPv4.
>             - Only work with instance support ENI, refer to `Instance families <https://www.alibabacloud.com/help/doc-detail/25378.htm>`_.

Reviewer: We could take this chance to improve the readability of this block, would you mind fixing it as a separate commit in your PR?

```suggestion
            The AlibabaCloud ENI integration with Cilium is subject to the following limitations:

            - It is currently only enabled for IPv4.
            - It only works with instances supporting ENI. Refer to `Instance families <https://www.alibabacloud.com/help/doc-detail/25378.htm>`_ for details.
```

Author: Done
